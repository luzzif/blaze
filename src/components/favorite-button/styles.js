export const styles = {
    root: {
        width: 28,
        height: 28,
        "& svg": {
            fontSize: 20
        }
    }
};
