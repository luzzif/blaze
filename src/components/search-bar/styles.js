export const styles = {
    root: {
        margin: "0 8px",
        width: "100%"
    },
    content: {
        padding: "0 8px !important"
    },
    container: {
        display: "flex"
    },
    input: {
        border: "none",
        outline: "none",
        width: "100%",
        height: "28px",
        fontSize: "12px",
        flexGrow: "1"
    }
};
