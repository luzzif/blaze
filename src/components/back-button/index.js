import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { ArrowBack } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicBackButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <ArrowBack />
    </IconButton>
);

BasicBackButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const BackButton = enhanceComponent(BasicBackButton);
