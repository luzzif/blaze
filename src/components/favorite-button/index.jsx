import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { Favorite } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicFavoriteButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <Favorite />
    </IconButton>
);

BasicFavoriteButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const FavoriteButton = enhanceComponent(BasicFavoriteButton);
