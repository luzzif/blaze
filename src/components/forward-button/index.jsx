import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { ArrowForward } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicForwardButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <ArrowForward />
    </IconButton>
);

BasicForwardButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const ForwardButton = enhanceComponent(BasicForwardButton);
