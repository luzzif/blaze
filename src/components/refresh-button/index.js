import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { Refresh } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicRefreshButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <Refresh />
    </IconButton>
);

BasicRefreshButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const RefreshButton = enhanceComponent(BasicRefreshButton);
