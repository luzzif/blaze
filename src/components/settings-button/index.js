import React from "react";
import PropTypes from "prop-types";
import { IconButton, withStyles } from "@material-ui/core";
import { Menu } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";

const BasicSettingsButton = ({ classes, ...rest }) => (
    <IconButton {...rest} classes={{ root: classes.root }}>
        <Menu />
    </IconButton>
);

BasicSettingsButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const SettingsButton = enhanceComponent(BasicSettingsButton);
