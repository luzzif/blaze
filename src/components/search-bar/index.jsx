import React from "react";
import PropTypes from "prop-types";
import { Card, CardContent, withStyles, Grid } from "@material-ui/core";
import { compose } from "redux";
import { styles } from "./styles";
import { FavoriteButton } from "../favorite-button";

const BasicSearchBar = ({ classes, ...rest }) => (
    <Card {...rest} classes={{ root: classes.root }}>
        <CardContent classes={{ root: classes.content }}>
            <div className={classes.container}>
                <input className={classes.input} type="text" />
                <FavoriteButton />
            </div>
        </CardContent>
    </Card>
);

BasicSearchBar.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const SearchBar = enhanceComponent(BasicSearchBar);
