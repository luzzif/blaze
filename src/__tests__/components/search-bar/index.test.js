import React from "react";
import { SearchBar } from "./../../../components/search-bar";
import TestRenderer from "react-test-renderer";
import { FavoriteButton } from "../../../components/favorite-button";

describe("search bar", () => {
    let testInstance = null;

    beforeEach(() => {
        testInstance = TestRenderer.create(<SearchBar />);
    });

    it("should be rendered", () => {
        expect(testInstance.toJSON().type).toBe("div");
        expect(testInstance).toMatchSnapshot();
    });

    it("should contain an input of type text", () => {
        const inputElement = testInstance.root.findByType("input");
        expect(inputElement).toBeTruthy();
        expect(inputElement.props.type).toBe("text");
        expect(testInstance).toMatchSnapshot();
    });

    it("should contain a favorite button", () => {
        expect(testInstance.root.findByType(FavoriteButton)).toBeTruthy();
        expect(testInstance).toMatchSnapshot();
    });
});
