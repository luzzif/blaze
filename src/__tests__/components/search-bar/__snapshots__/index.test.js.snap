exports[`search bar should be rendered 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation1-8 MuiPaper-rounded-6 MuiCard-root-4 BasicSearchBar-root-1"
  style={
    Object {
      "width": "auto",
    }
  }>
  <div
    className="MuiCardContent-root-32 BasicSearchBar-content-2">
    <div
      className="MuiGrid-container-33 MuiGrid-align-items-xs-center-41">
      <div
        className="MuiGrid-item-34 MuiGrid-grid-md-11-100">
        <input
          className="BasicSearchBar-input-3"
          type="text" />
      </div>
      <div
        className="MuiGrid-item-34 MuiGrid-grid-md-1-90">
        <div
          className="MuiGrid-container-33 MuiGrid-justify-xs-flex-end-51">
          <div
            className="MuiGrid-item-34">
            <button
              className="MuiButtonBase-root-137 MuiIconButton-root-131 BasicFavoriteButton-root-130"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button">
              <span
                className="MuiIconButton-label-136">
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-140"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24">
                  <path
                    d="M0 0h24v24H0z"
                    fill="none" />
                  <path
                    d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z" />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-147" />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`search bar should contain a favorite button 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation1-8 MuiPaper-rounded-6 MuiCard-root-4 BasicSearchBar-root-1"
  style={
    Object {
      "width": "auto",
    }
  }>
  <div
    className="MuiCardContent-root-32 BasicSearchBar-content-2">
    <div
      className="MuiGrid-container-33 MuiGrid-align-items-xs-center-41">
      <div
        className="MuiGrid-item-34 MuiGrid-grid-md-11-100">
        <input
          className="BasicSearchBar-input-3"
          type="text" />
      </div>
      <div
        className="MuiGrid-item-34 MuiGrid-grid-md-1-90">
        <div
          className="MuiGrid-container-33 MuiGrid-justify-xs-flex-end-51">
          <div
            className="MuiGrid-item-34">
            <button
              className="MuiButtonBase-root-137 MuiIconButton-root-131 BasicFavoriteButton-root-130"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button">
              <span
                className="MuiIconButton-label-136">
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-140"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24">
                  <path
                    d="M0 0h24v24H0z"
                    fill="none" />
                  <path
                    d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z" />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-147" />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`search bar should contain an input of type text 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation1-8 MuiPaper-rounded-6 MuiCard-root-4 BasicSearchBar-root-1"
  style={
    Object {
      "width": "auto",
    }
  }>
  <div
    className="MuiCardContent-root-32 BasicSearchBar-content-2">
    <div
      className="MuiGrid-container-33 MuiGrid-align-items-xs-center-41">
      <div
        className="MuiGrid-item-34 MuiGrid-grid-md-11-100">
        <input
          className="BasicSearchBar-input-3"
          type="text" />
      </div>
      <div
        className="MuiGrid-item-34 MuiGrid-grid-md-1-90">
        <div
          className="MuiGrid-container-33 MuiGrid-justify-xs-flex-end-51">
          <div
            className="MuiGrid-item-34">
            <button
              className="MuiButtonBase-root-137 MuiIconButton-root-131 BasicFavoriteButton-root-130"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button">
              <span
                className="MuiIconButton-label-136">
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-140"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24">
                  <path
                    d="M0 0h24v24H0z"
                    fill="none" />
                  <path
                    d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z" />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-147" />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
